### PR TITLE
Fix local dpl installation when cwd is not $TRAVIS_BUILD_DIR

### PR DESCRIPTION
**Pull request by BanzaiMan**
When execution gets to deployment and PWD is not `$TRAVIS_BUILD_DIR`, installing `dpl` with the locally built `dpl-*.gem` file fails.

This PR resolves this issue.

Failure with cf6d29c9c: https://travis-ci.org/BanzaiMan/travis_production_test/builds/333491097#L540
Success with fc63b337: https://staging.travis-ci.org/BanzaiMan/travis_staging_test/jobs/704197#L533

### CODE REVIEW

File: lib/travis/build/addons/deploy/script.rb [hunk 1]
```diff
@@ -142,12 +142,12 @@ def run
 
             def install
               sh.if "$(rvm use $(travis_internal_ruby) do ruby -e \"puts RUBY_VERSION\") = 1.9*" do
-                cmd(dpl_install_command(WANT_18), echo: false, assert: !allow_failure, timing: true)
+                cmd(dpl_install_command(WANT_18), echo: true, assert: !allow_failure, timing: true)
```
Comment (cotsog) [inline lib/travis/build/addons/deploy/script.rb]: Is echoing enabled on purpose here or it's a remainder of your tests?

File: lib/travis/build/addons/deploy/script.rb [hunk 2]
```diff
@@ -142,12 +142,12 @@ def run
...
               end
               sh.else do
-                cmd(dpl_install_command, echo: false, assert: !allow_failure, timing: true)
+                cmd(dpl_install_command, echo: true, assert: !allow_failure, timing: true)
```
Comment (cotsog) [inline lib/travis/build/addons/deploy/script.rb]: Same as above.

File: lib/travis/build/addons/deploy/script.rb (hunk 1 above)
Comment (BanzaiMan) [reply lib/travis/build/addons/deploy/script.rb]: It's done on purpose; so that it is a little clearer what is happening.

**Comment by bnferguson** [comment]
Seeing problems in Enterprise too, but don't quite look like this. More like; 

```
ERROR:  While executing gem ... (OptionParser::AmbiguousOption)

    ambiguous option: -*.gem

The command "rvm $(travis_internal_ruby) --fuzzy do ruby -S gem install dpl -v '< 1.9' -*.gem --local --pre" failed and exited with 1 during .
```

Looks like we may need to roll another release with updated `travis-build`?  Thoughts?

**Comment by BanzaiMan** [comment]
@bnferguson For the record: I think that is a particular bug where `travis_internal_ruby` returns `1.9*` and `edge` is defined in the config. Our hosted services do not do this, so we never saw it in production. At any rate, the error should be fixed with this PR.

**Comment by bnferguson** [comment]
Ahhh thanks for the clarification. Good to know why we didn't see it in hosted before.

**Pull request merged**